src/utils/barcode.ts

```diff
@@ -0,0 +1,81 @@
+import * as bwipjs from "bwip-js";
+import { tryCatch } from "fp-ts/lib/TaskEither";
+import * as JsBarcode from "jsbarcode";
+import { DOMImplementation, XMLSerializer } from "xmldom";
+
+export interface IBarcodeOutput {
+  png: string;
+  pngMimeType: string;
+  svg: string;
+  svgMimeType: string;
+}
+
+interface IBwipOptions {
+  bcid: string;
+  text: string;
+  includetext: boolean;
+}
+
+const toBase64Png = (options: IBwipOptions) => {
+  return tryCatch(
+    () => {
+      return new Promise<string>((resolve, reject) =>
+        bwipjs.toBuffer(options, (err, png) => {
```

Comment:
I believe you can use `util.promisify` to convert the callback to a promise

Reply:
or better [taskify](https://github.com/gcanti/fp-ts/blob/1.17.0/src/TaskEither.ts#L325)